src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -173,22 +175,36 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                         AddPolicyProperty(type, options);
 
                         Type objectType;
-                        if (IsNativelySupportedCollection(type))
-                        {
-                            // Use the type from the property policy to get any late-bound concrete types (from an interface like IDictionary).
+                        if (PolicyProperty.DeclaredPropertyType.IsInterface)
                             objectType = PolicyProperty.RuntimePropertyType;
-                        }
                         else
-                        {
-                            // We need to create the declared instance for types implementing natively supported collections.
                             objectType = PolicyProperty.DeclaredPropertyType;
-                        }
 
-                        CreateObject = options.MemberAccessorStrategy.CreateConstructor(objectType);
+                        CreateObject = options.MemberAccessorStrategy.CreateConstructor(objectType)
```

Comment:
We should leave this as null and throw at the site of deserialization.

Reply:
It's called all over so to reduce code duplication I encapsulated the throw into a method on JsonClassInfo. Please let me know what you think of that.